spec/request-reply-promoter.spec.js

```diff
@@ -56,19 +56,6 @@ describe('function promoter =>', () => {
         result({"$order": [source]}, {"$order": [streamingOutput]});
     });
 
-    describe('when called with functions with hooks => ', () => {
```

Comment:
this test is still valid, as the hooks should be preserved through promotion

Reply:
it does not matter at that point, hooks will have been executed so there is no point keeping them.

Reply:
right. I'm still used to promotion happening for the process rather than for an invocation.